.github/workflows/release.yml

```diff
@@ -0,0 +1,30 @@
+name: Release gem on RubyGems.org
+
+on:
+  workflow_dispatch:
```

Comment:
Is there a special reason why we are making these `workflow_dispatch` as opposed to being trigged on a special `v*` tag or when a `release` is published like in [here](https://github.com/Shopify/thread_safety/commit/a94e9aebcbb13193302302ba6c23a651ebc742e8)?

I think with a release published, we can even ask another team member/repo member to approve the release publishing, so that we have more security over the process than running a workflow.